app/classes/DB.php

```diff
@@ -39,33 +40,35 @@ class DB {
     /**
      * Constructor of the DB class. Calls the methods to load environment variables and establish a connection to the database.
      */
-    public function __construct(bool $autoconnect = true, bool $autoclose = false) {
+    public function __construct(bool $autoConnect = true, private readonly bool $autoClose = false)
+    {
         $this->sqlPath = cml_config('SQL_PATH');
-        $this->autoclose = $autoclose;
-        if ($autoconnect) {
+        if ($autoConnect) {
             $this->connected = true;
-            $this->connect(cml_config('DB_HOST'), cml_config('DB_USER'), cml_config('DB_PASSWORD'), cml_config('DB_NAME')); 
+            $this->connect(cml_config('DB_HOST'), cml_config('DB_USER'), cml_config('DB_PASSWORD'), cml_config('DB_NAME'));
         }
     }
 
-    public function __destruct() {
-        if ($this->autoclose === true) {
+    public function __destruct()
+    {
+        if ($this->autoClose === true) {
             $this->close();
         }
     }
-    
+
     /**
      * Establishes a connection to the database.
      *
      * @param string $host The database host.
      * @param string $user The database username.
      * @param string $pass The database password.
-     * @param string $dbname The database name.
+     * @param string $dbName
      */
-    public function connect(string $host, string $user, string $pass, string $dbname) {
-        $this->conn = @new \mysqli($host, $user, $pass, $dbname);
+    public function connect(string $host, string $user, string $pass, string $dbName): void
```

Comment:
I don't like to return void. If there is no return type, it's always void :). I mean, we can leave it there, but just saying.